toolchain/check/cpp_thunk.cpp

```diff
@@ -328,8 +328,8 @@ auto PerformCppThunkCall(Context& context, SemIR::LocId loc_id,
   CARBON_CHECK(thunk_function_params.size() == num_params);
   CARBON_CHECK(callee_arg_ids.size() == num_params);
   llvm::SmallVector<SemIR::InstId> thunk_arg_ids;
-  thunk_arg_ids.reserve(callee_arg_ids.size());
-  for (size_t i = 0; i < callee_function_params.size(); ++i) {
+  thunk_arg_ids.reserve(num_params);
+  for (size_t i = 0; i < num_params; ++i) {
```

Comment:
Two minor thoughts glancing at this, you might consider seq or zip:

```suggestion
  for (auto i : llvm::seq(num_params)) {
```

```suggestion
  for (auto [callee_param, thunk_param, arg_id] : llvm::zip(callee_function_params, thunk_function_params, calle_arg_ids)) {
```

Reply:
Thanks!
Using `llvm::zip()`.